### ISSUE TITLE
Relicense under dual MIT/Apache-2.0

### ISSUE THREAD

**Issue by cmr** [andelf/rust-sdl2_ttf#26]
# Why?

The MIT license requires reproducing countless copies of the same copyright
header with different names in the copyright field, for every MIT library in
use. The Apache license does not have this drawback, and has protections from
patent trolls and an explicit contribution licensing clause. However, the
Apache license is incompatible with GPLv2. This is why Rust is dual-licensed as
MIT/Apache (the "primary" license being Apache, MIT only for GPLv2 compat), and
doing so would be wise for this project. This also makes this crate suitable
for inclusion in the Rust standard distribution and other project using dual
MIT/Apache.

# How?

To do this, get explicit approval from each contributor of copyrightable work
(as not all contributions qualify for copyright) and then add the following to
your README:

```
## License

Licensed under either of
 * Apache License, Version 2.0 ([LICENSE-APACHE](LICENSE-APACHE) or http://www.apache.org/licenses/LICENSE-2.0)
 * MIT license ([LICENSE-MIT](LICENSE-MIT) or http://opensource.org/licenses/MIT)
at your option.

### Contribution

Unless you explicitly state otherwise, any contribution intentionally submitted
for inclusion in the work by you shall be dual licensed as above, without any
additional terms or conditions.
```

and in your license headers, use the following boilerplate (based on that used in Rust):

```
// Copyright (c) 2015 t developers
// Licensed under the Apache License, Version 2.0
// <LICENSE-APACHE or
// http://www.apache.org/licenses/LICENSE-2.0> or the MIT
// license <LICENSE-MIT or http://opensource.org/licenses/MIT>,
// at your option. All files in the project carrying such
// notice may not be copied, modified, or distributed except
// according to those terms.
```

And don't forget to update the `license` metadata in your `Cargo.toml`!

# Contributor checkoff

 - [ ] @andelf
 - [ ] @sbidin
 - [ ] @bvssvni
 - [ ] @Outfl3sh
 - [ ] @crumblingstatue
 - [ ] @dpetersen
 - [ ] @jadpole
 - [ ] @TheNeikos
 - [ ] @cristicbz
 - [ ] @dpc
 - [ ] @caryhaynie
 - [ ] @Revertron
 - [ ] @werecat

**Comment by bvssvni** [comment]
I approve dual licensing under the MIT/Apache.

**Comment by dpetersen** [comment]
The new license is fine with me.

**Comment by cristicbz** [comment]
I agree with my contributions being dual licensed under Apache2.0 and MIT

**Comment by werecat** [comment]
I approve of the dual license

**Comment by dpc** [comment]
Fine with me.

**Comment by dpc** [comment]
Out of curiosity. I'm licensing my projects under MPLv2, and I wonder if it has any drawback, comparing to BSD/Apache.

**Comment by cmr** [comment]
It's a copyleft license, so it's in a different category from these permissive licenses, but is overall still excellent.

On January 8, 2016 9:58:18 PM EST, "Dawid Ciężarkiewicz" <notifications@github.com> wrote:
>Out of curiosity. I'm licensing my projects under MPLv2, and I wonder
>if it has any drawback, comparing to BSD/Apache.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/andelf/rust-sdl2_ttf/issues/26#issuecomment-170184466

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by TheNeikos** [comment]
I agree dual licensing as well.

**Comment by Outfl3sh** [comment]
I contributed like one line, but I agree, if it's necessary.

**Comment by sbidin** [comment]
Sure, I agree.

**Comment by crumblingstatue** [comment]
Either way is fine by me, but if my consent is required, I provide it.

**Comment by jadpole** [comment]
You have my approval.
